_posts/2022-09-06-conscryptmode.md

```diff
@@ -0,0 +1,19 @@
+To be more consistent with Android, Robolectric needed to be updated to use Conscrypt as the default security provider.
+ 
+I was already well familiar with the technologies used in the project such as Java, Kotlin, Android Studio, Git and GitHub. I was interested in Android Developement and I was looking for an android project that uses Java as the primary language. I found Robolectric as a perfect project to contribute according to my skills and I was ready to learn about testing concepts since Robolectric is a popular unit testing framework for Android. 
+
+At first, we thought we could switch Robolectric from BouncyCastle to Conscrypt directly. We quickly found we can't completely switch Robolectric to Conscrypt because there were some internal tests that needed BouncyCastle as a fallback security provider. After we added it in, we decided to create a ConscryptMode annotation that lets users choose to use Conscrypt or not. Robolectric can now use either Conscrypt and BouncyCastle or just BouncyCastle as the security provider. Robolectric will search for a requested security feature from Conscrypt first. If it does not support it, Robolectric will try BouncyCastle second.
+
+- If ConscryptMode is `ON`, it will install Conscrypt and BouncyCastle.
+- If ConscryptMode is `OFF`, it will only install BouncyCastle.
+
+I have learnt about unit testing like how to test whether a piece of code is working according to the expected behavior or not, writing tests for specific scenario, providing self-explanatory names to methods so that other developers can understand it easily in future.
+
+I came to know how to debug large codebase like Robolectric and how to find the main cause behind an error/test case failure. I learnt about annotations like defining the annotation and how it works when we use it somewhere in the project, and how compiler record these annotations to provide that information at times whenever needed. 
+
+According to me, when we contribute to an open-source project, we always need to take care of few things like how the changes can affect other end users/projects who depends on it and sometimes there may be cases when we can't fix/resolve issues because our project also depends on some other library/project, and what we are left with is to wait for a new update with hope that it fixes those errors. Like in our case, we used Conscrypt but it doesn't have support for Mac M1. We were left with two options either create our own custom Conscrypt or wait for an update from the official Conscrypt and hope that it will fix the problems. Therefore, we should be aware of the impact that project can have in both positive and negative aspects. So that, we can find a best solution which doesn't affect the end users in a wrong way.
```

Comment:
Split the first sentence into a few separate sentences. Ex.

``` 
When we contribute to an open-source project, we always need to check how our changes can affect other end users/projects that depend on it. Sometimes there may be cases where we can't fix/resolve issues because our project also depends on some other library/project. That might mean we are left to wait for a new update and hope that it fixes those errors.
```